src/utils/extractAllowedOptionsUpdates.ts

```diff
@@ -0,0 +1,37 @@
+import {isUnknownObject} from './guards';
+import {isEqual} from './isEqual';
+
+export type UnknownOptions = {[k: string]: unknown};
+
+export const extractAllowedOptionsUpdates = (
+  options: unknown | void,
+  prevOptions: unknown | void,
+  immutableKeys: string[]
+): UnknownOptions | null => {
+  if (!isUnknownObject(options) || !isUnknownObject(prevOptions)) {
```

Comment:
In the current version of React Stripe.js, you can update options from `null` to a non-null value. It's not allowed by the types but currently works. I think this guard here prevents that from happening.

Failing test case for `createElementComponent.test.tsx` that currently passes on `master`:

```tsx
it('updates the Element when options change from null to non-null value', () => {
  const {rerender} = render(
    <Elements stripe={mockStripe}>
      {/* @ts-expect-error */}
      <CardElement options={null} />
    </Elements>
  );

  rerender(
    <Elements stripe={mockStripe}>
      <CardElement options={{style: {base: {fontSize: '30px'}}}} />
    </Elements>
  );

  expect(mockElement.update).toHaveBeenCalledWith({
    style: {base: {fontSize: '30px'}},
  });
});
```